bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java

```diff
@@ -384,6 +385,41 @@ public void addAttributes(Collection<BlackboardAttribute> attributes) throws Tsk
 		attrsCache.addAll(attributes);
 	}
 
+	/**
+	 * Adds a collection of attributes to this artifact in a single operation
+	 * (faster than adding each attribute individually). This function also
+	 * executes in the inserts within the transaction context.
+	 *
+	 * @param attributes		      The collection of attributes.
+	 * @param caseDbTransaction the transaction in the scope of which the
+	 *                          operation is to be performed, managed by the
+	 *                          caller
+	 *
+	 * @throws TskCoreException If an error occurs and the attributes were not
+	 *                          added to the artifact.
+	 */
+	public void addAttributes(Collection<BlackboardAttribute> attributes, final SleuthkitCase.CaseDbTransaction caseDbTransaction) throws TskCoreException {
+
+		if (attributes.isEmpty()) {
+			return;
+		}
```

Comment:
We should probably throw an IllegalArgumentException if the collection is empty, since that is a programming error and we are doing the caller no favors by allowing it. This requires changing the conditional on line 497 in Blackboard.

Reply:
@rcordovano Did you mean a TSK exception or really mean an uncaught IllegalArgumentException?

Reply:
I meant IllegalArgumentException. I sent you an email with the rationale.